segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/extendeds3/ExtendedS3StorageTest.java

```diff
@@ -0,0 +1,266 @@
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+import org.gaul.s3proxy.AuthenticationType;
+import org.gaul.s3proxy.S3Proxy;
+import org.jclouds.ContextBuilder;
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.BlobStoreContext;
+import org.jclouds.logging.slf4j.config.SLF4JLoggingModule;
+import org.junit.After;
+import org.junit.Before;
+
+/**
+ * Unit tests for ExtendedS3Storage.
+ */
+@Slf4j
+public class ExtendedS3StorageTest extends IdempotentStorageTestBase {
+    private static final String BUCKET_NAME = "pravegatest";
+    private ExtendedS3StorageFactory storageFactory;
+    private ExtendedS3StorageConfig adapterConfig;
+    private S3JerseyClient client = null;
+    private S3Proxy s3Proxy;
+    private final ConcurrentMap<String, AclSize> aclMap = new ConcurrentHashMap<>();
+    private final String endpoint = "http://127.0.0.1:" + TestUtils.getAvailableListenPort();
+
+    @Before
+    public void setUp() throws Exception {
+
+        URI uri = URI.create(endpoint);
+        Properties properties = new Properties();
+        properties.setProperty("s3proxy.authorization", "none");
+        properties.setProperty("s3proxy.endpoint", endpoint);
+        properties.setProperty("jclouds.provider", "filesystem");
+        properties.setProperty("jclouds.filesystem.basedir", "/tmp/s3proxy");
+
+        ContextBuilder builder = ContextBuilder
+                .newBuilder("filesystem")
+                .credentials("x", "x")
+                .modules(ImmutableList.<Module>of(new SLF4JLoggingModule()))
+                .overrides(properties);
+        BlobStoreContext context = builder.build(BlobStoreContext.class);
+        BlobStore blobStore = context.getBlobStore();
+
+        s3Proxy = S3Proxy.builder().awsAuthentication( AuthenticationType.AWS_V2_OR_V4, "x", "x")
+                .endpoint(uri)
+                .keyStore("", "")
+                .blobStore(blobStore)
+                .ignoreUnknownHeaders(true)
+                .build();
+
+        s3Proxy.start();
+
+        this.adapterConfig = ExtendedS3StorageConfig.builder()
+                                                    .with(ExtendedS3StorageConfig.BUCKET, BUCKET_NAME)
+                                                    .with(ExtendedS3StorageConfig.ACCESS_KEY_ID, "x")
+                                                    .with(ExtendedS3StorageConfig.SECRET_KEY, "x")
+                                                    .with(ExtendedS3StorageConfig.ROOT, "test")
+                                                    .with(ExtendedS3StorageConfig.URI, endpoint)
+                                                    .build();
+        createStorage();
+
+        try {
+            client.createBucket(BUCKET_NAME);
+        } catch (S3Exception e) {
+            if (!e.getErrorCode().equals("BucketAlreadyOwnedByYou")) {
```

Comment:
Is it possible that in the case the bucket has been created previously that it is dirty and could affect the test results? I'm wondering if we should fail in this case rather than ignore.